### ISSUE TITLE
Review `import-x/extensions` config

### ISSUE THREAD

**Issue by fregante** [xojs/eslint-config-xo#119]
I was getting `import-x/extensions` in some non-TS files and zero extension warning in TS files, making me wonder what was up. Looking at the `import-x/extensions` mentions in the repo, it seems that it's waiting for something, but this config is already working for me in TS:

```js
'import-x/extensions': ['error', {
	js: 'ignorePackages',
	json: 'always',
}],
```

Likely not comprehensive though.

https://github.com/refined-github/refined-github/blob/74334c5e341e9a84a3723c62f0d881859533dcba/eslint.config.js#L104-L107

**Comment by sindresorhus** [comment]
The reason TS got zero enforcement is that we disabled both `import-x/extensions` and `n/file-extension-in-import` for TypeScript, since both resolved `./foo.js` to the real `./foo.ts` and checked the extension against the `.ts`, which breaks the "write `.js` for a `.ts` source" convention.

That's no longer true for `n/file-extension-in-import` — `eslint-plugin-n` now maps the resolved `.ts`→`.js` (tsconfig-aware, and adapts to `allowImportingTsExtensions`) and has an autofix, so it correctly enforces the `.js` convention. `import-x/extensions` still can't do this, so it stays off for TS.

Your suggested config only silences the false errors (`ts` defaults to `never`, so nothing gets enforced on TS-to-TS imports); re-enabling `n/file-extension-in-import` actually enforces + autofixes it, which is what we want.

**Comment by fregante** [comment]
Great! What would be the future advantage of `import-x/extension` if `n/file-extension-in-import` already covers all?

**Comment by sindresorhus** [comment]
That's something I would look at at such time, not now.

**Comment by fregante** [comment]
I only ask because the comment still mentions "Disabled until it supports this" rather than "already covered by n/file-extension-in-import", but 👍

**Comment by fregante** [comment]
Sorry to bug you but `import-x/extension` is still enabled in this config. It shows up in my .svelte files suggesting I change the imports' extension to .ts, even if I don't have that option enabled

https://github.com/search?q=repo%3Axojs%2Feslint-config-xo%20import-x%2Fextension&type=code

**Comment by sindresorhus** [comment]
https://github.com/xojs/eslint-config-xo/releases/tag/v0.58.1

**Comment by fregante** [comment]
Dōmo arigatō, Mr. Culaudo!
